opsdroid/connector/matrix/connector.py

```diff
@@ -223,20 +247,35 @@ async def get_nick(self, roomid, mxid):
         setting.
         """
         if self.room_specific_nicks:
-            try:
-                return await self.connection.get_room_displayname(roomid, mxid)
-            except Exception:  # pylint: disable=W0703
-                # Fallback to the non-room specific one
-                logging.exception("Failed to lookup room specific nick for %s.", mxid)
+            if roomid is None:
+                return mxid
```

Comment:
I would move this up into the top if statement, and fall back to not-roomspecific nicks if room is `None`.
```suggestion
```